expertise/execute_expertise.py

```diff
@@ -0,0 +1,177 @@
+            publications_path=None,
+            submissions_path=None,
+            scores_path=Path(config['model_params']['scores_path']).joinpath(config['name'] + '.csv')
+        )
+
+        if config['model_params'].get('sparse_value'):
+            mfr_predictor.sparse_scores(
+                scores_path=Path(config['model_params']['scores_path']).joinpath(config['name'] + '_sparse.csv')
+            )
+
+    if config['model'] == 'specter+mfr':
+        from .models import multifacet_recommender
+        ens_predictor = multifacet_recommender.EnsembleModel(
+            specter_dir=config['model_params'].get('specter_dir', "./models/multifacet_recommender/specter/"),
+            mfr_feature_vocab_file=config['model_params'].get('mfr_feature_vocab_file', "./feature/dictionary_index"),
+            mfr_checkpoint_dir=config['model_params'].get('mfr_checkpoint_dir', "./"),
+            mfr_epochs=config['model_params'].get('mfr_epochs', 100),
+            work_dir=config['model_params'].get('work_dir', "./"),
+            average_score=config['model_params'].get('average_score', False),
+            max_score=config['model_params'].get('max_score', True),
+            specter_batch_size=config['model_params'].get('specter_batch_size', 16),
+            mfr_batch_size=config['model_params'].get('mfr_batch_size', 50),
+            merge_alpha=config['model_params'].get('merge_alpha', 0.8),
+            use_cuda=config['model_params'].get('use_cuda', False),
+            sparse_value=config['model_params'].get('sparse_value')
+        )
+        ens_predictor.set_archives_dataset(archives_dataset)
+        ens_predictor.set_submissions_dataset(submissions_dataset)
+        ens_predictor.embed_publications(
+            specter_publications_path=Path(config['model_params']['publications_path']).joinpath('pub2vec.jsonl'),
+            mfr_publications_path=None, skip_specter=config['model_params'].get('skip_specter', False))
+        ens_predictor.embed_submissions(
+            specter_submissions_path=Path(config['model_params']['submissions_path']).joinpath('sub2vec.jsonl'),
+            mfr_submissions_path=None, skip_specter=config['model_params'].get('skip_specter', False))
+        ens_predictor.all_scores(
+            specter_publications_path=Path(config['model_params']['publications_path']).joinpath('pub2vec.jsonl'),
+            mfr_publications_path=None,
+            specter_submissions_path=Path(config['model_params']['submissions_path']).joinpath('sub2vec.jsonl'),
+            mfr_submissions_path=None,
+            scores_path=Path(config['model_params']['scores_path']).joinpath(config['name'] + '.csv')
+        )
+
+        if config['model_params'].get('sparse_value'):
+            ens_predictor.sparse_scores(
+                scores_path=Path(config['model_params']['scores_path']).joinpath(config['name'] + '_sparse.csv')
+            )
+
+def execute_create_dataset(client, config_file=None, model_config=None):
```

Comment:
is this functions create the dataset or running the model? I'm confused here.

Reply:
In this file, the two functions `execute_expertise` corresponds to running the model, and `execute_create_dataset` corresponds to fetching data from OpenReview.

So everything from the following lines and below are moved to `execute_expertise`:
https://github.com/openreview/openreview-expertise/blob/b973858ce6434e3bbcfc473cda132dedaea36858/expertise/run.py#L12-L14

And everything from the following line and below is moved to `execute_create_dataset`:
https://github.com/openreview/openreview-expertise/blob/b973858ce6434e3bbcfc473cda132dedaea36858/expertise/create_dataset.py#L361

Which is used to define the work done by the queues:
https://github.com/openreview/openreview-expertise/blob/a86bddea5e48d9efd20d00f8fe1a0f1f1be7e0f8/expertise/service/or_queue.py#L80-L82
https://github.com/openreview/openreview-expertise/blob/a86bddea5e48d9efd20d00f8fe1a0f1f1be7e0f8/expertise/service/or_queue.py#L158-L164

Reply:
Got it! thanks for the explanation.